lib/active_merchant/billing/gateways/shift4_v2.rb

```diff
@@ -11,6 +11,28 @@ def credit(money, payment, options = {})
         commit('credits', post, options)
       end
 
+      def store(payment_method, options = {})
+        post = case payment_method
+               when CreditCard
+                 cc = {}.tap { |card| add_creditcard(card, payment_method, options) }[:card]
+                 options[:customer_id].blank? ? { email: options[:email], card: cc } : cc
+               when Check
+                 bank_account_object(payment_method, options)
+               else
+                 raise ArgumentError.new("Unhandled payment method #{payment_method.class}.")
+               end
+
+        commit url_for_store(payment_method, options), post, options
+      end
+
+      def url_for_store(payment_method, options = {})
+        case payment_method
```

Comment:
matter of taste maybe, but since we have a simple binary condition for payment_method we can use if.
```suggestion
   if payment_method.is_a?(CreditCard)
       options[:customer_id].blank?...
    elsif payment_method.is_a?(Check)
        'payment-methods'
    end
```